lxd/daemon.go

```diff
@@ -949,6 +1006,73 @@ func (d *Daemon) Stop() error {
 	return err
 }
 
+// Setup the API web server
+func (d *Daemon) setupWebServer() {
+	/* Setup the web server */
+	d.mux = mux.NewRouter()
+	d.mux.StrictSlash(false)
+
+	d.mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		SyncResponse(true, []string{"/1.0"}).Render(w)
+	})
+
+	for _, c := range api10 {
+		d.createCmd("1.0", c)
+	}
+
+	for _, c := range apiInternal {
+		d.createCmd("internal", c)
+	}
+
+	d.mux.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger.Info("Sending top level 404", log.Ctx{"url": r.URL})
+		w.Header().Set("Content-Type", "application/json")
+		NotFound.Render(w)
+	})
+
+}
+
+// Setup external authentication
+func (d *Daemon) setupExternalAuthentication(authEndpoint string) error {
+	if authEndpoint == "" {
+		d.externalAuth = nil
+		logger.Infof("External authentication disabled")
+		return nil
+	}
+
+	idmClient, err := idmclient.New(idmclient.NewParams{
+		BaseURL: authEndpoint,
+	})
+	if err != nil {
+		return err
+	}
+	key, err := bakery.GenerateKey()
+	if err != nil {
+		return err
+	}
+	pkLocator := httpbakery.NewThirdPartyLocator(nil, nil)
+	pkLocator.AllowInsecure()
```

Comment:
What does that mean?

Reply:
This allows untrusted HTTPS certs, I think I can drop it now (I used to test with https and a self-signed cert)